src/lib/storage/vector_compression/simd_bp128/simd_bp128_decompressor.hpp

```diff
@@ -69,24 +69,24 @@ class SimdBp128Decompressor : public BaseVectorDecompressor {
   size_t size() const final { return _size; }
 
  private:
-  bool _is_index_within_cached_block(size_t index) {
+  bool _is_index_within_cached_block(size_t index) const {
     const auto begin = _cached_block_first_index;
     const auto end = _cached_block_first_index + Packing::block_size;
     return begin <= index && index < end;
   }
 
   size_t _index_within_cached_block(size_t index) { return index - _cached_block_first_index; }
 
-  bool _is_index_within_cached_meta_block(size_t index) {
+  bool _is_index_within_cached_meta_block(size_t index) const {
     const auto begin = _cached_meta_block_first_index;
     const auto end = _cached_meta_block_first_index + Packing::meta_block_size;
     return begin <= index && index < end;
   }
 
-  size_t _index_relative_to_cached_meta_block(size_t index) { return index - _cached_meta_block_first_index; }
+  size_t _index_relative_to_cached_meta_block(size_t index) const { return index - _cached_meta_block_first_index; }
 
-  bool _is_index_after_cached_meta_block(size_t index) {
-    return (_cached_meta_block_first_index + Packing::meta_block_size) <= index;
```

Comment:
`_cached_meta_block_first_index ` was initialised with `std::numeric_limits<size_t>::max()`. So the first time this method is executed, `Packing::meta_block_size` was added to the maximum value of `size_t`, which led to an integer overflow.